### PR TITLE
mail-filter/postforward: Initial ebuild

### PR DESCRIPTION
**Pull request by Philippe23**
- Create initial ebuild
 - Work around upstream not putting their .go files in src/ which
   makes "go build" angry
 - Work around golang-build complaining about pkg not existing
   and just do the install by hand
 - Patch default sendmail path to respect EPREFIX and use the
   standard postfix install location instead of depending on
   it being in the path

Package-Manager: Portage-2.3.8, Repoman-2.3.3

### CODE REVIEW

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: Please convert it to EAPI 6.

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+inherit eutils golang-build
+
+DESCRIPTION="Postfix Sender Rewriting Scheme forwarding agent"
+#EGO_PN="github.com/zoni/${PN}"
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: Remove this line please.

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+EGO_PN="${PN}"
+SRC_URI="https://github.com/zoni/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+HOMEPAGE="https://github.com/zoni/postforward"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: This line can go.

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+DEPEND="
+	dev-lang/go:0"
+RDEPEND="
+	mail-filter/postsrsd"
+
+src_prepare() {
+	epatch "${FILESDIR}/${PN}_apply-sendmail-path.patch"
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: Define a PATCH array outside this function in which you'll add this file, like so:
```
PATCH=( "${FILESDIR}/${PN}_apply-sendmail-path.patch" )
```

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	sed -i -e "s/@GENTOO_PORTAGE_EPREFIX@/${EPREFIX}/" *.go
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: missing `|| die`

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+	# go build assumes files will be in src dir, but
+	# source files are in root in this package, so move
+	# them.
+	NEW_SRC_DIR="${S}/src/${EGO_PN}"
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: `export` ?

File: mail-filter/postforward/postforward-1.0.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	# adjust to it.
+	test ! -e "${NEW_SRC_DIR}" || die "${NEW_SRC_DIR} already exists"
+	mkdir -p "${NEW_SRC_DIR}" || die
+	mv *.go "${NEW_SRC_DIR}" || die
+}
+
+# Standard golang-build src_install complains about pkg not
+# existing, so we go custom.
+src_install() {
+	dodoc README.md
```
Comment (monsieurp) [inline mail-filter/postforward/postforward-1.0.1.ebuild]: call `einstalldocs` instead and remove this line.

**Comment by Philippe23** [comment]
Just wanted to say I'll probably get back to this later in the month.  Holidays, work, and a trip are getting in the way of me having a chance to work on this.

File: mail-filter/postforward/postforward-1.0.1.ebuild (hunk 6 above)
Comment (Philippe23) [reply mail-filter/postforward/postforward-1.0.1.ebuild]: Is `local` kosher?

File: mail-filter/postforward/postforward-1.0.1.ebuild (hunk 7 above)
Comment (Philippe23) [reply mail-filter/postforward/postforward-1.0.1.ebuild]: Fine to just call `default` instead?

Comment (Philippe23) [reply mail-filter/postforward/postforward-1.0.1.ebuild]: Nix that question, `default` calls `make install` which doesn't work for this package.  Forgot the whole reason I added the custom `src_install`....

**Comment by monsieurp** [comment]
Sure.

**Comment by mgorny** [comment]
Ping².

**Comment by Philippe23** [comment]
Updated to implement @monsieurp's requests.  Assumed he wanted the changes squashed into the original from the "improve the commit message" statement.  Also updated version to 1.1.0 since that has been released in the interim.

Build log: [postforward-1.1.0_20180313-153638.log](https://github.com/gentoo/gentoo/files/1807610/postforward-1.1.0_20180313-153638.log)

Gentoo bug: https://bugs.gentoo.org/650398

File: mail-filter/postforward/Manifest [hunk 8]
```diff
@@ -0,0 +1 @@
+DIST postforward-1.1.0.tar.gz 5707 SHA256 a2968dbc6374715d503c4efba67ee587d33a045a747d74253c4dd9f43df6fffa SHA512 39d7a33c454b1529a691b24fd69b311ba368caf59168d39146654cfe319ebc3cf73835e66e6c95a43747f389a6b954ac8da0d7e934e562c16cb85dc20c0f4804 WHIRLPOOL 329c42f424497f3b6dee433eab775752c3e59d37a8139f1e9f39eef9e749971a94321b98c88c6bd76144003c39116a26cf1386b7796e913494bffa689f3f6dd5
```
Comment (mgorny) [inline mail-filter/postforward/Manifest]: Those are wrong hashes. Please remove the Manifest file and make `repoman manifest` regenerate them.

File: mail-filter/postforward/postforward-1.1.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+src_prepare() {
+	default
+
+	# Dynamically fix EPREFIX lines made by ${PN}_apply-sendmail-path.patch
+	sed -i -e "s/@GENTOO_PORTAGE_EPREFIX@/${EPREFIX}/" *.go || die
+
+	# go build assumes files will be in src dir, but
+	# source files are in root in this package, so move
+	# them.
+	local NEW_SRC_DIR="${S}/src/${EGO_PN}"
```
Comment (mgorny) [inline mail-filter/postforward/postforward-1.1.0.ebuild]: The usual convention is to name global variables in uppercase but local variables in lowercase.

File: mail-filter/postforward/postforward-1.1.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+	# Freak out if there's already something there because
+	# it means the package has changed and we'll need to
+	# adjust to it.
+	test ! -e "${NEW_SRC_DIR}" || die "${NEW_SRC_DIR} already exists"
```
Comment (mgorny) [inline mail-filter/postforward/postforward-1.1.0.ebuild]: Don't use `test` in bash, use `[[ ... ]]`. Also note that `[[ ... ]]` normally doesn't require quoting.

**Comment by Philippe23** [comment]
Rebuilt manifest, changed case of local variable, and switched away from `test` per @mgorny's request.  Left quotes on for consistency because they're needed in the other two uses and don't hurt in the `[[ ]]` use case.

**Pull request closed without merging**